tests/utils/kudo/kudo.go

```diff
@@ -293,10 +294,12 @@ func UpdateInstanceParameters(
 		return err
 	}
 
-	err = WaitForOperatorDeployInProgress(namespaceName, instanceName)
-	if err != nil {
-		log.Errorf("Error waiting for operator deploy to be in-progress: %s", err)
-		return err
+	if waitForDeploy {
```

Comment:
The "wait for deploy complete" should also be inside the conditional block, no?

Reply:
Mmmmm, maybe. It only assures that the plan status is in 'COMPLETE', which is ok.

Maybe the parameter should be named `waitForDeployInProgress`, because that's what I tried to accomplish.

The problem with the "WaitForOperatorDeployInProgress" is that for an update to a ConfigMap, the "IN_PROGRESS" step is so short that the test misses it.

Reply:
The idea of waiting first for the deploy to be "in-progress" and then waiting for "completed" is to follow the deploy state machine. A parameter update will trigger a deploy, which will cause the deploy status to go from "complete" -> "in-progress" -> "complete".

So the a parameter named "waitForDeploy" waiting for "in-progress" and then "complete" makes sense, because that's what one would want from a function that updates an instances parameters: change the parameter and wait for the parameter change to be reflected fully, i.e., for the deploy plan to complete.

> The problem with the "WaitForOperatorDeployInProgress" is that for an update to a ConfigMap, the "IN_PROGRESS" step is so short that the test misses it.

Is this reproducible? Where did you see it happening?

Reply:
Yes, generally I would agree with the COMPLETE->IN_PROGRESS->COMPLETE flow.

I've noticed the problem in this step:
https://github.com/mesosphere/kudo-cassandra-operator/pull/46/files#diff-013e06c2e358068fd6cff2d5724007a7R31

In this update, only an additional service is deployed, without updating the Pods of the StatefulSet. I assume that the Apply of the Service is quick enough that the window for "IN_PROGRESS" is very short and not reliable enough to wait for it